forge-game/src/main/java/forge/game/cost/CostAdjustment.java

```diff
@@ -128,14 +128,10 @@ private static void applyRaise(final Cost cost, final SpellAbility sa, final Sta
         } else if (st.hasParam("Amount")) {
             String amount = st.getParam("Amount");
             if ("Escalate".equals(amount)) {
-                SpellAbility sub = sa;
-                while (sub != null) {
-                    if (sub.getDirectSVars().containsKey("CharmOrder")) {
-                        count++;
-                    }
-                    sub = sub.getSubAbility();
+                SpellAbility tail = sa.getTailAbility();
```

Comment:
I don't think this works:
if one mode is effects A -> B -> C your logic will not find it?

Reply:
CharmOrder is set for sorting, so the Bottom of the SpellAbility Chain should have the highest CharmOrder

using getTailAbility to get to them Bottom, and using getSVar, sould look up for the next SpellAbility with CharmOrder param and return it

Reply:
you will only check the direct parent B and miss the SVar on A

Reply:
I really don't see how calling `C.hasSVar` would reach A:

```
    @Override
    public boolean hasSVar(final String name) {
        return sVars.containsKey(name) || findSVar(name).isPresent();
    }
```

```
    protected Optional<IHasSVars> findSVar(final String name) {
        return getSVarFallback().stream().filter(f -> f.hasSVar(name)).findFirst();
    }
```

```
    protected List<IHasSVars> getSVarFallback() {
        List<IHasSVars> result = Lists.newArrayList();
        if (getParent() != null) {
            result.add(getParent());
        }
        result.addAll(super.getSVarFallback());
        return result;
    }
```

Reply:
C.hasSVar calls B.hasSVar which calls A.hasSVar

C.hasSVar fails to find it in sVars
then calls C.findSVar which calls C.getSVarFallback
which calls C.getParent to get B

then it calls B.hasSVar fails to find it in sVars
then calls B.findSVar which calls B.getSVarFallback
which calls B.getParent to get A

then it calls A.hasSVar which finds it in sVars